RustFxA/FxAWebViewModel.swift

```diff
@@ -0,0 +1,308 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import WebKit
+import Foundation
+import Account
+import MozillaAppServices
+import Shared
+import SwiftKeychainWrapper
+
+
+extension URL {
+    var toRequest: URLRequest {
+        return URLRequest(url: self)
+    }
+    /// Figuring out if two urls are the same based on scheme, host and path.
+    func isEquivilent(to url: URL) -> Bool {
+        return self.scheme == url.scheme &&
+        self.host == url.host &&
+        self.path == url.path
+    }
+}
+
+enum FxAPageType {
+    case emailLoginFlow
+    case qrCode(url: String)
+    case settingsPage
+}
+
+// See https://mozilla.github.io/ecosystem-platform/docs/fxa-engineering/fxa-webchannel-protocol
+// For details on message types.
+fileprivate enum RemoteCommand: String {
+    //case canLinkAccount = "can_link_account"
+    // case loaded = "fxaccounts:loaded"
+    case status = "fxaccounts:fxa_status"
+    case login = "fxaccounts:oauth_login"
+    case changePassword = "fxaccounts:change_password"
+    case signOut = "fxaccounts:logout"
+    case deleteAccount = "fxaccounts:delete_account"
+    case profileChanged = "profile:change"
+}
+
+typealias STATE = (URLRequest, UnifiedTelemetry.EventMethod?)
+
+class FxAWebViewModel {
+    
+    fileprivate let pageType: FxAPageType
+    fileprivate let profile: Profile
+    fileprivate let firefoxAccounts: RustFirefoxAccounts
+    fileprivate let leanPlumClient: LeanPlumClient
+    
+    // This is not shown full-screen, use mobile UA
+    static let MobileUserAgent = UserAgent.mobileUserAgent()
```

Comment:
The static variable should also be lowerCamelCase.

Reply:
Done